test/Infrastructure.IntegrationTest/Services/Implementations/EfMigrationTesterService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Data;
+using Bit.Core.Enums;
+using Bit.Infrastructure.EntityFramework.Repositories;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using MySqlConnector;
+using Npgsql;
+
+namespace Bit.Infrastructure.IntegrationTest.Services;
+
+public class EfMigrationTesterService : IMigrationTesterService
+{
+    private readonly DatabaseContext _databaseContext;
+    private readonly SupportedDatabaseProviders _databaseType;
+    private readonly string _migrationName;
+
+    public EfMigrationTesterService(
+        DatabaseContext databaseContext,
+        SupportedDatabaseProviders databaseType,
+        string migrationName)
+    {
+        _databaseContext = databaseContext;
+        _databaseType = databaseType;
+        _migrationName = migrationName;
+    }
+
+    public void ApplyMigration()
+    {
+        // Delete the migration history to ensure the migration is applied
+        DeleteMigrationHistory();
+
+        var migrator = _databaseContext.GetService<IMigrator>();
+        migrator.Migrate(_migrationName);
+    }
+
+    private void DeleteMigrationHistory()
+    {
+        var deleteCommand = "DELETE FROM __EFMigrationsHistory WHERE MigrationId LIKE @migrationName";
+        IDbDataParameter? parameter = null;
+
+        switch (_databaseType)
+        {
+            case SupportedDatabaseProviders.MySql:
+                parameter = new MySqlParameter("@migrationName", "%" + _migrationName);
+                break;
+            case SupportedDatabaseProviders.Postgres:
+                deleteCommand = "DELETE FROM \"__EFMigrationsHistory\" WHERE \"MigrationId\" LIKE @migrationName";
+                parameter = new NpgsqlParameter("@migrationName", "%" + _migrationName);
+                break;
+            case SupportedDatabaseProviders.Sqlite:
+                parameter = new SqliteParameter("@migrationName", "%" + _migrationName);
+                break;
+        }
+
+        _databaseContext.Database.ExecuteSqlRaw(deleteCommand, parameter);
+    }
```

Comment:
I tried that now but it requires that we add the `HistoryRow` class as part of the `DbContext` and I don't think we should have that table accessible, right?

Reply:
Yeah, I think you are right @r-tome, I saw [this](https://github.com/dotnet/efcore/blob/78400fcec978162059afbf744775d3db720820ca/src/EFCore.Relational/Migrations/HistoryRepository.cs#L218-L224) and thought it might be implicitly part of the model but now that I look a bit closer they are creating their own `ModelBuilder` for that [here](https://github.com/dotnet/efcore/blob/78400fcec978162059afbf744775d3db720820ca/src/EFCore.Relational/Migrations/HistoryRepository.cs#L89-L95). 

I can definitely live with how it is now, was just hoping we could make it a little more future proof.